cmd/spire-server/cli/run/run.go

```diff
@@ -56,6 +56,7 @@ type serverConfig struct {
 	SVIDTTL             string             `hcl:"svid_ttl"`
 	TrustDomain         string             `hcl:"trust_domain"`
 	UpstreamBundle      bool               `hcl:"upstream_bundle"`
+	RegistrationPruning string             `hcl:"registration_pruning"`
```

Comment:
so would you have everything as-is, **but**: 
- cli sends hard-coded value instead of taking configuration
- documentation of available config is removed
?

Reply:
Documentation removed, CLI and related config bits removed, we can define the constant directly in manager.go replacing `_warningCadence`. Of course we can also remove all the logic related to detecting various suboptimal cadence config values.

@azdagron also noticed that the Expiry field on registration entries is not currently indexed... I would imagine that would be important too as part of this work...

Reply:
As I am a bit wary of forcing a particular cadence for a new feature on existing users, I will be making a hard-coded cadence much larger, probably 2 hours. This seems like a good amount of time for any special use cases we may not be thinking of.
Of course this is moot for any registration entry **without** expiry set, as those will never be pruned, and documentation will be updated as such on registration API.

Let me know if 2 hours seems too long or too short.

Reply:
> As I am a bit wary of forcing a particular cadence for a new feature on existing users

Is there something in particular you're worried about?

> Let me know if 2 hours seems too long or too short.

2 hours feels quite long to me - I was thinking something along the lines of five minutes. A user that sets an expiry on the minutes scale will almost certainly be surprised that it takes 2 hours to actually expire.

Reply:
> A user that sets an expiry on the minutes scale will almost certainly be surprised that it takes 2 hours to actually expire.
In the issue we discussed that expectations should be managed about automated pruning; it is not meant to be a safety net of any kind for security but instead a nice-to-have storage management feature. SPIRE should never be issuing identities based on expired registrations, whether they have been pruned or not.

> Is there something in particular you're worried about?
I'm am worried about the unknown scenarios for users' SPIRE deployments. IMO, storage is an easier ask on a consumer than processing power, so we should, through a longer cadence, have lower tax on processing and accept that storage fills up a bit more.
That being said 2 hours is probably a bit paranoid. I will change to 5 minutes.